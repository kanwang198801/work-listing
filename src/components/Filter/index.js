import React from 'react';
import PropTypes from 'prop-types';
import { Radio, Button } from 'antd';
import styles from './styles.module.css';

const Filter = ({ title, onChange, onReset, value, options, id }) => {
  return (
    <div className={styles.filter} id={id}>
      <div>
        <h3>{title}</h3>
        <Button onClick={onReset}> Reset</Button>
      </div>
      <Radio.Group onChange={onChange} value={value}>
        {options.map((option) => (
          <Radio key={option} value={option}>
            {option}
          </Radio>
        ))}
      </Radio.Group>
    </div>
  );
};

Filter.propTypes = {
  title: PropTypes.string.isRequired,
  onChange: PropTypes.func.isRequired,
  onReset: PropTypes.func.isRequired,
  options: PropTypes.arrayOf(PropTypes.string).isRequired,
  value: PropTypes.string.isRequired,
  id: PropTypes.string.isRequired
};

export default Filter;
