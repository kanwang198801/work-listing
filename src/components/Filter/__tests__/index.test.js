import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import renderer from 'react-test-renderer';
import Filter from '../index';

const onFilterClick = jest.fn();
const onReset = jest.fn();
const filter = (
  <Filter
    title="test filter"
    onChange={onFilterClick}
    onReset={onReset}
    value=""
    options={['test1', 'test2']}
    id="test-filter"
  />
);
describe('Filter Test Suite', () => {
  it('snapshot', () => {
    const tree = renderer.create(filter).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders the Filter component', () => {
    const component = render(filter);
    expect(component).toBeTruthy();
  });

  it('Reset button should be clickable', () => {
    render(filter);
    fireEvent.click(document.querySelector('#test-filter-reset-button'));
    expect(onReset).toHaveBeenCalledTimes(1);
  });

  it('Reset button should be clickable', () => {
    render(filter);
    fireEvent.click(document.querySelector('.ant-radio-input'));
    expect(onFilterClick).toHaveBeenCalledTimes(1);
  });
});
