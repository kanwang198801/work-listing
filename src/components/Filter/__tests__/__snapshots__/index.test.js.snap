// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Filter Test Suite snapshot 1`] = `
<div
  className="filter"
  id="test-filter"
>
  <div>
    <h3>
      test filter
    </h3>
    <button
      className="ant-btn"
      id="test-filter-reset-button"
      onClick={[Function]}
      type="button"
    >
      <span>
        Reset
      </span>
    </button>
  </div>
  <div
    className="ant-radio-group ant-radio-group-outline"
  >
    <label
      className="ant-radio-wrapper"
    >
      <span
        className="ant-radio"
        style={Object {}}
      >
        <input
          checked={false}
          className="ant-radio-input"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyPress={[Function]}
          onKeyUp={[Function]}
          type="radio"
          value="test1"
        />
        <span
          className="ant-radio-inner"
        />
      </span>
      <span>
        test1
      </span>
    </label>
    <label
      className="ant-radio-wrapper"
    >
      <span
        className="ant-radio"
        style={Object {}}
      >
        <input
          checked={false}
          className="ant-radio-input"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyPress={[Function]}
          onKeyUp={[Function]}
          type="radio"
          value="test2"
        />
        <span
          className="ant-radio-inner"
        />
      </span>
      <span>
        test2
      </span>
    </label>
  </div>
</div>
`;
